src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java

```diff
@@ -48,53 +38,22 @@
 
 @Extension
 public class SCMListenerOctaneImpl extends SCMListener {
-	private static final Logger logger = LogManager.getLogger(SCMListenerOctaneImpl.class);
-	private static final DTOFactory dtoFactory = DTOFactory.getInstance();
 
-	@Override
-	public void onChangeLogParsed(Run<?, ?> run, SCM scm, TaskListener listener, ChangeLogSet<?> changelog) throws Exception {
-		super.onChangeLogParsed(run, scm, listener, changelog);
-
-		SCMProcessor scmProcessor = SCMProcessors.getAppropriate(scm.getClass().getName());
-		if (scmProcessor == null) {
-			logger.debug("no processors found for SCM provider of type '" + scm.getType() + "', SCM data won't be extracted");
-			return;
-		}
-
-		try {
-			SCMData scmData = extractSCMData(run, scm, scmProcessor);
-			if (scmData != null) {
-				CIEvent event = createSCMEvent(run, scmData);
-				CIJenkinsServicesImpl.publishEventToRelevantClients(event);
-			}
-		} catch (Throwable throwable) {
-			logger.error("failed to build and/or dispatch SCM event for " + run, throwable);
-		}
-	}
-
-	private SCMData extractSCMData(Run run, SCM scm, SCMProcessor scmProcessor) {
-		SCMData result = null;
-		if (run.getParent() instanceof MatrixConfiguration || run instanceof AbstractBuild) {
-			AbstractBuild build = (AbstractBuild) run;
-			if (!build.getChangeSet().isEmptySet()) {
-				result = scmProcessor.getSCMData(build, scm);
-			}
-		} else if (run instanceof WorkflowRun) {
-			WorkflowRun wRun = (WorkflowRun) run;
-			if (!wRun.getChangeSets().isEmpty()) {
-				result = scmProcessor.getSCMData(wRun, scm);
-			}
-		}
-		return result;
-	}
-
-	private CIEvent createSCMEvent(Run<?, ?> run, SCMData scmData) {
-		return dtoFactory.newDTO(CIEvent.class)
-				.setEventType(CIEventType.SCM)
-				.setProject(BuildHandlerUtils.getJobCiId(run))
-				.setBuildCiId(BuildHandlerUtils.getBuildCiId(run))
-				.setCauses(CIEventCausesFactory.processCauses(run))
-				.setNumber(String.valueOf(run.getNumber()))
-				.setScmData(scmData);
-	}
+    @Override
+    public void onChangeLogParsed(Run<?, ?> run, SCM scm, TaskListener listener, ChangeLogSet<?> changelog) throws Exception {
+        super.onChangeLogParsed(run, scm, listener, changelog);
+
+        if (!skipScmEvent(run)) {
+            CIEvent scmEvent = CIEventFactory.createScmEvent(run, scm);
+            if (scmEvent != null) {
+                CIJenkinsServicesImpl.publishEventToRelevantClients(scmEvent);
+            }
+        }
+    }
+
+    private static boolean skipScmEvent(Run<?, ?> run) {
+        boolean isMultibranchJob = run.getParent().getParent() != null &&
```

Comment:
Is the first getParent can return null as well?

Reply:
No, first  parent of the run - is job itself, it  cannot be null